agent/nmap_agent.py

```diff
@@ -372,6 +372,19 @@ def _emit_fingerprints(
                 else:
                     raise ValueError(f"Incorrect ip version {version}")
 
+                if host.get("os") is not None:
+                    if host.get("os").get("osmatch", {}) is not None:
+                        logger.debug("sending results to selector %s", selector)
+
+                        os = host.get("os").get("osmatch", {})[0]
```

Comment:
```suggestion
                        os = host.get("os").get("osmatch")[0]
```
since we know that `get('osmatch') is not None from earlier check + `{}[0]` will give us a KeyError anyway